ml/backend/ggml/ggml.go

```diff
@@ -299,10 +302,15 @@ func New(r *os.File, params ml.BackendParams) (ml.Backend, error) {
 
 	// concurrently read in tensor data. uses a section reader which is safe for concurrent reads
 	sr := io.NewSectionReader(r, int64(meta.Tensors().Offset), n-int64(meta.Tensors().Offset))
-	var g errgroup.Group
+
+	var doneCount atomic.Uint32
+
+	g, ctx := errgroup.WithContext(context.Background())
```

Comment:
This would spin up a considerable number of goroutines for large models. Should we limit it to `g.SetLimit(runtime.NumCPU())`?